### PR TITLE
PP-586: Added to PTL tests.

### PR DESCRIPTION
**Pull request by vccardenas**
#### Issue-ID

* **[PP-586](https://pbspro.atlassian.net/browse/PP-586)  On a Cray X-series, create a vnode per compute node (adding PTL tests)**

#### Problem description
* Added to PTL tests.

#### Cause / Analysis
* The test was missing the setting $vnode_per_numa_node False.

#### Solution description
* Compare the pbsnodes output when vnode_per_numa_node was unset versus when vnode_per_numa_node was set to False.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_cray_vnode_per_numa.py [hunk 1]
```diff
@@ -128,6 +128,31 @@ def test_settings(self):
...
+        # versus when vnode_per_numa_node was set to False.
+
+        if (len(vnodes_combined) == len(vnodes_combined1)):
+            self.logger.info(
+                "pbsnodes outputs are equal in length.")
+            for n in vnodes_combined:
+                if n not in vnodes_combined1:
+                    self.logger.error(
+                        "ERROR vnode %s has differing element." % n['id'])
+                    self.assertTrue(False)
```
Comment (lisa-altair) [inline test/tests/functional/pbs_cray_vnode_per_numa.py]: These self.assertTrue(False) statements get the job done, but they seem awkward to me.  Is there a better way to do this?

**Comment by anamikau** [comment]
Hey Lisa,

What is awkward? It is same as if (!(True)) being used in C or other languages.

-Anamika

**Comment by mike0042** [comment]
I think the use of self.assertTrue(False) is a pretty common practice in PTL. I know I have used it.

**Comment by lisa-altair** [comment]
Yeah, apparently I've used it to.  :)  The reason it seems awkward is because we are forcing the assert to happen, we aren't really "checking" anything.  I brought it up because when I asked a couple other people they said they don't like to do something like assertTrue(False)...which I took to mean there might be a better way...Maybe there isn't...Maybe we can add to the PTL framework (as a separate enhancement) a more generic function that exits on problem.

**Comment by arungrover** [comment]
I think we need to have a mechanism of raising a generic test failure exception when the test writer thinks that the test case has failed.
something like - raise TestFailedException(msg="Test failed because of xyz reason")
I don't see something similar currently supported in ptl which makes us rely on asserts.


In any case, @vccardenas the change you made is testing whether the default behavior is same as the switch being turned off.
You might either want to change it as a different test case (which I think it is) OR you might want to change the doc string of the test case itself to reflect what test case is trying to test.

**Comment by arungrover** [comment]
looks good @vccardenas. As soon as you have another sign-off please rebase and squash your commits.

**Comment by hirenvadalia** [comment]
Just a question: Is PR title right?

**Comment by vccardenas** [comment]
My changes are ready to be merged into the golden repository.  Please take this forward.

**Pull request merged**